docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -120,10 +128,118 @@ The reasons that DNS service is per-network instead of per-space are:
 * VPC networks are isolated by nature. It will be difficult to have a shared DNS pod that's accessible to pods in various VPC networks.
 
 ### Service
-For VPC-isolated networking, service IP support has to be provided w/o relying on iptable/IPVS rules on host netns. In other words, traditional kube-proxy can not used in VPC-isolated networking.
 
+#### Type definition
+
+Addition of service spec is metadata.network field. By default, it is the default network of the tenant. 
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  network: my-network
+  selector:
+    app: MyApp
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+```
+
+#### Naming Problem & Solution
+
+Services, as one of the critical network related resource types, are withing the network boundary. There is no such *shared* services of tenant or whole cluster, as both the critical virtual IP address and the associated endpoints (typically leading to pods) are all of same network.
+
+Putting service type in tenant level is out of primacy as it has following flaws:
+1. service IP address has to be the same across all networks, which implies some range of service IP needs to be predefined as shared across networks of a tenant (tenant manageability burden);  
+1. associating network-agnostic service to network-specific endpoints would introduce significant inconsistency;
+1. kubectl query for service (e.g. ```kubectl describe service```) has no way to return meaningful content of endpoints.
```

Comment:
How about Kubernetes' endpoints, how the object details will change?

Reply:
endpoints will be in later design - we expect the minimum change of type (with addition of spec.network), hopefully it can keep the change impact small.